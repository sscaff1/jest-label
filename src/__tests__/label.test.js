// MOCK types and traits
const label = require('../label');
const traits = {
  SLOW: 'Slow'
};
const types = {
  INTEGRATION: 'Integration',
  UNIT: 'Unit'
};

describe('label', () => {
  describe(label({ name: 'should be successfully applied', type: types.INTEGRATION }), () => {
    const testMockLabel = { name: 'mock' };
    const arrayOfTests = [
      [{ ...testMockLabel, type: types.UNIT }],
      [{ ...testMockLabel, type: types.INTEGRATION, traits: traits.SLOW }],
      [{ ...testMockLabel, type: types.INTEGRATION, traits: [traits.SLOW] }]
    ];

    test.each(arrayOfTests)('for %p should match snapshot', labelPassed => {
      expect(label(labelPassed)).toMatchSnapshot();
    });
  });

  describe(label({ name: 'should throw an error when attempted', type: types.INTEGRATION }), () => {
    const testMockLabel = { name: 'mock' };
    const arrayOfTests = [
      [{ ...testMockLabel }],
      [{ ...testMockLabel, type: 'fake' }],
      [{ ...testMockLabel, type: null }],
      [{ ...testMockLabel, type: types.UNIT, traits: [] }],
      [{ ...testMockLabel, type: types.UNIT, traits: ['__fake__'] }],
      [{ ...testMockLabel, type: types.UNIT, traits: '__fake__' }]
    ];

    test.each(arrayOfTests)('for %p should match error snapshot', labelPassed => {
      expect(() => label(labelPassed)).toThrowErrorMatchingSnapshot();
    });
  });
});
