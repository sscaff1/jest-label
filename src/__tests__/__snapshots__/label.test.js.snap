// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label (Test Type: Integration) should be successfully applied for {"name": "mock", "traits": "Slow", "type": "Integration"} should match snapshot 1`] = `"(Test Type: Integration, Test Traits: Slow) mock"`;

exports[`label (Test Type: Integration) should be successfully applied for {"name": "mock", "traits": [Array], "type": "Integration"} should match snapshot 1`] = `"(Test Type: Integration, Test Traits: Slow) mock"`;

exports[`label (Test Type: Integration) should be successfully applied for {"name": "mock", "type": "Unit"} should match snapshot 1`] = `"(Test Type: Unit) mock"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock", "traits": "__fake__", "type": "Unit"} should match error snapshot 1`] = `"You passed invalid traits"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock", "traits": [Array], "type": "Unit"} should match error snapshot 1`] = `"You passed invalid traits"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock", "traits": [Array], "type": "Unit"} should match error snapshot 2`] = `"You passed invalid traits"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock", "type": "fake"} should match error snapshot 1`] = `"All tests must be labeled with a valid type"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock", "type": null} should match error snapshot 1`] = `"All tests must be labeled with a valid type"`;

exports[`label (Test Type: Integration) should throw an error when attempted for {"name": "mock"} should match error snapshot 1`] = `"All tests must be labeled with a valid type"`;
